gcd/src/test/java/org/spine3/server/storage/datastore/DatastoreWrapperShould.java

```diff
@@ -64,18 +75,79 @@ public void fail_to_start_transaction_if_one_is_active() {
 
     @Test(expected = IllegalStateException.class)
     public void fail_to_finish_not_active_transaction() {
-        final DatastoreWrapper wrapper = DatastoreWrapper.wrap(Given.localDatastore());
+        final DatastoreWrapper wrapper = DatastoreWrapper.wrap(Given.testDatastore());
         wrapper.startTransaction();
         assertTrue(wrapper.isTransactionActive());
         wrapper.commitTransaction();
         assertFalse(wrapper.isTransactionActive());
         wrapper.rollbackTransaction();
     }
 
+    // Some variables act in different cases and should have self-explainatory names
+    @SuppressWarnings("UnnecessaryLocalVariable")
+    @Test
+    public void support_big_bulk_reads() throws InterruptedException {
+        final int bulkSize = 1001;
+        final int maxWritesPerCall = 500;
+        final int firstPartStart = 0;
+        final int firstPartEnd = firstPartStart + maxWritesPerCall;
+        final int secondPartStart = firstPartEnd;
+        final int secondPartEnd = secondPartStart + maxWritesPerCall;
+        final int thirdPartStart = secondPartEnd;
+        final int thirdPartEnd = bulkSize;
+
+
+        final TestDatastoreWrapper wrapper = TestDatastoreWrapper.wrap(Given.testDatastore(), false);
+        final Map<Key, Entity> entities = Given.nEntities(bulkSize, wrapper);
+        final List<Entity> sourceEntities = new ArrayList<>(entities.values());
+
+        final Entity[] firstPart = new Entity[maxWritesPerCall];
+        final Entity[] secondPart = new Entity[maxWritesPerCall];
+        final Entity[] thirdPart = new Entity[1]; // Only 1 element left. Change this when changing bulkSize
+        sourceEntities.subList(firstPartStart, firstPartEnd).toArray(firstPart);
+        sourceEntities.subList(secondPartStart, secondPartEnd).toArray(secondPart);
+        sourceEntities.subList(thirdPartStart, thirdPartEnd).toArray(thirdPart);
+
+        wrapper.createOrUpdate(firstPart);
+        wrapper.createOrUpdate(secondPart);
+        wrapper.createOrUpdate(thirdPart);
+
+        // Wait some time to make sure the writing is complete
```

Comment:
`Wait for some time ...`. And the dot at the end, since you started this comment as a sentence.